include/stlcache/stlcache.hpp

```diff
@@ -557,6 +562,7 @@ namespace stlcache {
          * \see size 
          */
         size_type max_size() const throw() {
+            shared_guard<mutex> lock(mtx);
```

Comment:
As i understand, we do not change _maxEntries after construction, so no lock is needed